### PR TITLE
Update Error Message and Validation Process for Calories

### LINKED ISSUE #150: [PE-D] Calories Validation
Edge case testing - negative numbers for calories gives error message: “Calories should be at least 1 digit long, or must be an integer”, even though provided -100 fulfils these requirements

Might need another kind of error message

To reproduce:
add n/running d/10 mins at/31-12-2020 c/-100 m/chest,arm t/home t/gym

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: chrystalquek/ped#8

### LINKED ISSUE #151: [PE-D] Add exercise with very high calorie burn does not update graph
Top graph did not update upon input calories = 999999999999

Steps to reproduce:
add n/flop d/10 mins at/24-10-2020 c/999999999999 m/chest,arm t/home t/gym

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: chrystalquek/ped#9

### LINKED ISSUE #152: [PE-D] Adding exercises of very high calories consecutively does not update result display and clear text field
To reproduce:

1) add n/flop d/10 mins at/24-10-2020 c/999999999999 m/chest,arm t/home t/gym

2) add n/flop d/10 mins at/24-10-2020 c/999999999999 m/chest,arm t/home t/gym

(step 2 gives expected response)

3) add n/flopp d/10 mins at/24-10-2020 c/999999999999 m/chest,arm t/home t/gym

(step 3 is unexpected - while exercise is added, it gives wrong error message and does not clear out text field)

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: chrystalquek/ped#10

### LINKED ISSUE #154: [PE-D] Clear command does not clear templates
Would have expected templates to clear.

Previous:

After 'clear':

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: chrystalquek/ped#12

### PR DESCRIPTION
**Pull request by Nahoyhp**
Fixes #150

### CODE REVIEW

**Comment by Nahoyhp** [comment]
Will Fix it tmr

File: src/main/java/seedu/address/ui/CaloriesGraph.java [hunk 1]
```diff
@@ -37,11 +37,11 @@ public CaloriesGraph(HashMap<String, Integer> dayCalories) {
     /**
      * Generates the graph based on this HashMap that is assigned during initialisation.
      */
+    @SuppressWarnings("unchecked")
```
Comment (aaronlwm) [inline src/main/java/seedu/address/ui/CaloriesGraph.java]: suppresswarnings

File: src/test/java/seedu/address/model/exercise/CaloriesTest.java [hunk 2]
```diff
@@ -15,20 +15,20 @@ public void constructor_invalidCalories_throwsIllegalArgumentException() {
...
     @Test
     public void isValidCalories() {
-        // null calories
-        assertThrows(NullPointerException.class, () -> Calories.isValidCalories(null));
 
-        // invalid calories
+
+        // Boundary Testing
+        assertTrue(Calories.isValidCalories("0"));
+        assertTrue(Calories.isValidCalories("1000"));
```
Comment (aaronlwm) [inline src/test/java/seedu/address/model/exercise/CaloriesTest.java]: hardcoded

File: src/test/java/seedu/address/testutil/TypicalExercise.java [hunk 3]
```diff
@@ -37,19 +37,19 @@
             .withCalories(VALID_CALORIES_SIT_UP).withMuscleTags(VALID_MUSCLE_ARM).build();
 
     public static final Exercise RUN = new ExerciseBuilder().withName("Run")
-            .withDescription("Test 3").withDate("17-10-2020").withCalories("12348")
+            .withDescription("Test 3").withDate("17-10-2020").withCalories("123")
```
Comment (aaronlwm) [inline src/test/java/seedu/address/testutil/TypicalExercise.java]: hardcoded

**Pull request merged**